file-formats/csnm/type/zif_aff_csnm_v1.intf.abap

```diff
@@ -0,0 +1,43 @@
+"! ABAP File Format for CSN Model Transport Object Generic Editor
+INTERFACE zif_aff_csnm_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">CSN File</p>
+    "! CSN file
+    BEGIN OF ty_csn_file,
+      "! <p class="shorttext">CSN File Name</p>
+      "! CSN file name
+      name TYPE string,
+    END OF ty_csn_file.
+
+  "! <p class="shorttext">CSN Files</p>
+  "! CSN files
+  TYPES ty_csn_file_names TYPE STANDARD TABLE OF ty_csn_file WITH DEFAULT KEY.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">CSN File Names</p>
+      "! CSN file name
+      csn_file_names TYPE ty_csn_file_names,
+    END OF ty_general_information.
```

Comment:
I guess, I would remove general information in your case and have just a top level section `can_file_names` instead. 

Do you see any advantage if we keep the additional nesting?

Reply:
I do not see any advantage of the additional nesting, sinnce the csn_file_names is the only attribute..